consoles/sshVirtsh.pm

```diff
@@ -542,17 +560,32 @@ __END"
 
         # set default boot delay
         $self->run_cmd(qq{echo 'bios.bootDelay = "10000"' >> $vmx}, domain => 'sshVMwareServer');
-
-        # inject cloud init metadata and userdata required for the image if there are any
-        my $ci_meta = $bmwqemu::vars{CLOUD_INIT_META};
-        my $ci_user = $bmwqemu::vars{CLOUD_INIT_USER};
-        my $ci_encoding = $bmwqemu::vars{CLOUD_INIT_ENCODING};
-
-        if ($ci_meta && $ci_user && $ci_encoding) {
-            $self->run_cmd(qq{echo 'guestinfo.metadata = "$ci_meta"' >> $vmx}, domain => 'sshVMwareServer');
-            $self->run_cmd(qq{echo 'guestinfo.metadata.encoding = "$ci_encoding"' >> $vmx}, domain => 'sshVMwareServer');
-            $self->run_cmd(qq{echo 'guestinfo.userdata = "$ci_user"' >> $vmx}, domain => 'sshVMwareServer');
-            $self->run_cmd(qq{echo 'guestinfo.userdata.encoding = "$ci_encoding"' >> $vmx}, domain => 'sshVMwareServer');
+        my $fb_tool = $bmwqemu::vars{FIRST_BOOT_CONFIG};
+
+        if ($fb_tool && $fb_tool ne 'wizard') {
+            my $encoding = 'gzip+base64';
+            if ($fb_tool =~ /combustion|ignition/) {
+                if ($bmwqemu::vars{FIRST_BOOT_COMBUSTION}) {
+                    my $conf = $self->_encode_config($bmwqemu::vars{FIRST_BOOT_COMBUSTION}, 'FIRST_BOOT_COMBUSTION');
+                    $self->run_cmd(qq{echo 'guestinfo.combustion.encoding = "$encoding"' >> $vmx}, domain => 'sshVMwareServer');
```

Comment:
Doesn't exist

Reply:
Does it mean user cannot set the encoding or that I have missed `script` after `combustion`?

Reply:
> Does it mean an user cannot set the encoding

Correct

Reply:
Thanks, removed